### PR TITLE
fix: pnpm link --global <pkg> should not change the type of the dependency

### PR DESCRIPTION
**Pull request by lvqq**
Close https://github.com/pnpm/pnpm/issues/5478

### CODE REVIEW

File: packages/plugin-commands-installation/src/link.ts [hunk 1]
```diff
@@ -178,11 +183,14 @@ export async function handler (
       globalPkgNames = pkgNames
     }
     const globalPkgPath = pathAbsolute(opts.dir)
-    globalPkgNames.forEach((pkgName) => pkgPaths.push(path.join(globalPkgPath, 'node_modules', pkgName)))
+    globalPkgNames.forEach((pkgName) => {
+      const pkgPath = path.join(globalPkgPath, 'node_modules', pkgName)
+      pkgPaths.push(pkgPath)
+      const dependencyType = getDependencyTypeFromManifest(manifest, pkgName)
+      targetDependenciesFieldMap[pkgPath] = dependencyType ?? linkOpts.targetDependenciesField
```
Comment (zkochan) [inline packages/plugin-commands-installation/src/link.ts]: it'll be better if you pass the manifest to the `link()` function and get the depType inside the link function.

Comment (lvqq) [reply packages/plugin-commands-installation/src/link.ts]: That'll be better. Thanks

File: packages/types/src/misc.ts [hunk 2]
```diff
@@ -1,5 +1,7 @@
 export type DependenciesField = 'optionalDependencies' | 'dependencies' | 'devDependencies'
 
+export type DependenciesManifestField = DependenciesField | 'peerDependencies'
```
Comment (zkochan) [inline packages/types/src/misc.ts]: This name doesn't make sense. Maybe rename it to DependenciesAndPeersField

**Review by zkochan — APPROVED** [review]

**Comment by Anoesj** [comment]
❤️

**Pull request merged**